File: app/src/main/java/com/example/yuan/newsapp/NewsTitleAdapter.java
package com.example.yuan.newsapp;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ArrayAdapter;
import android.widget.TextView;

import java.util.List;

public class NewsTitleAdapter extends ArrayAdapter<NewsTitle> {

    private int resourceId;

    public NewsTitleAdapter(Context context, int resource,List<NewsTitle> objects) {
        super(context, resource, objects);
        resourceId = resource;
    }

    @Override
    public View getView(int position, View convertView, ViewGroup parent) {
        NewsTitle newsTitle = getItem(position);
        View view;
        ViewHolder viewHolder;
        if (convertView != null) {
            view = LayoutInflater.from(getContext()).inflate(resourceId, parent, false);
            viewHolder = new ViewHolder();
            viewHolder.title = view.findViewById(R.id.title);
            viewHolder.date = view.findViewById(R.id.date);
            viewHolder.category = view.findViewById(R.id.category);
            viewHolder.author_name = view.findViewById(R.id.author_name);
            view.setTag(viewHolder);
        } else {
            view = convertView;
            viewHolder = (ViewHolder) view.getTag();
        }
        viewHolder.title.setText(newsTitle.getTitle());
        viewHolder.date.setText(newsTitle.getDate());
        viewHolder.category.setText(newsTitle.getCategory());
        viewHolder.author_name.setText(newsTitle.getAuthor_name());
        return view;
    }

    class ViewHolder {
        TextView title;
        TextView date;
        TextView category;
        TextView author_name;
    }
}

File: app/src/main/java/com/yuan/news/adapter/NewsTitleAdapter.kt
package com.yuan.news.adapter

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ArrayAdapter
import android.widget.ImageView
import android.widget.TextView
import com.yuan.news.R
import com.yuan.news.bean.DataBean
import com.yuan.news.bean.NewsResult
import kotlinx.android.synthetic.main.item_title.view.*
import org.w3c.dom.Text

/**
 *yuan
 *2020/4/6
 **/
class NewsTitleAdapter(context: Context, resource: Int, objects: List<NewsResult.ResultBean.DataBean>): ArrayAdapter<NewsResult.ResultBean.DataBean>(context, resource, objects) {

    var resourceId: Int? = null
    var newsTitleList: List<NewsResult.ResultBean.DataBean>? = null

    init {
        resourceId = resource
        newsTitleList = objects
    }

    override fun getView(position: Int, convertView: View?, parent: ViewGroup): View {
        val dataBean = newsTitleList!![position]
        var viewHolder: ViewHolder?
        var view: View?
        if (convertView == null) {
            view = LayoutInflater.from(context).inflate(resourceId!!, parent, false)
            viewHolder = ViewHolder(view)
            view.tag = viewHolder
        } else {
            view = convertView
            viewHolder = view.tag as ViewHolder
        }
        viewHolder.title!!.text = dataBean.title
        viewHolder.source!!.text = dataBean.category
        viewHolder.date!!.text = dataBean.date
        return view!!
    }

    inner class ViewHolder(view: View) {
        var title: TextView? = null
        var source: TextView? = null
        var date: TextView? = null

        init {
            title = view.findViewById(R.id.newsTitleTextView)
            source = view.findViewById(R.id.sourceTextView)
            date = view.findViewById(R.id.dateTextView)
        }
    }
}